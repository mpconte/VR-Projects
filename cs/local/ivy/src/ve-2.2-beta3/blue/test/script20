#
# hash tests
#
set h [hash]
$h set foo 0
$h set bar 1
$h set joe 2
if { [$h get foo] != 0 || [$h get bar] != 1 || [$h get joe] != 2 } {
    error "basic set/get failed"
}
if { [$h get foobar] != "" } {
    error "get of non-existent failed"
}
[stdout] writeln [$h export]

# create new hash
set h [hash]
$h import {{foo 0} {bar 1} {joe 2}}
if { [$h get foo] != 0 || [$h get bar] != 1 || [$h get joe] != 2 } {
    error "basic set/get failed (2)"
}
if { [$h get foobar] != "" } {
    error "get of non-existent failed (2)"
}
[stdout] writeln [$h export]
