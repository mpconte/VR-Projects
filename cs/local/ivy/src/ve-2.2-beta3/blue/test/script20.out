{bar 1} {joe 2} {foo 0}
{bar 1} {joe 2} {foo 0}
result = ok
